### DISCUSSION TITLE
db creation error for the demo project

### DISCUSSION THREAD

**Discussion by yseraf** [outline/outline#6267] [Self Hosting]
Trying to setup a running instance of Outline (based on the [official documentation](https://docs.getoutline.com/s/hosting/doc/docker-7pfeLP5a8t)), I have multiple problems but finally get stuck at the db creation phase.

I use docker compose to run the instance. First, outline service can't access db because 127.0.0.1:5432 is not accessible. I had to replace localhost by service name (postgres). Maybe related to WSL ?

Passed this point, the oultine service is able to reach postgres but has an issue with "dialect" specified by database.json for db:create and db:drop commands.

**To Reproduce**
1. Follow the [documentation](https://docs.getoutline.com/s/hosting/doc/docker-7pfeLP5a8t) but replace in docker.env
    ```
    DATABASE_URL=postgres://user:pass@localhost:5432/outline
    DATABASE_URL_TEST=postgres://user:pass@localhost:5432/outline-test
    ```
    by
    ```
    DATABASE_URL=postgres://user:pass@postgres:5432/outline
    DATABASE_URL_TEST=postgres://user:pass@postgres:5432/outline-test
    ```
2. Run the command `docker-compose run --rm outline yarn db:create --env=production-ssl-disable`
3. Get the following error
    ```
    [+] Building 0.0s (0/0)
    [+] Creating 2/0
     ✔ Container outline-postgres-1  Running                                                                                            0.0s 
     ✔ Container outline-redis-1     Running                                                                                            0.0s 
    [+] Building 0.0s (0/0)
    yarn run v1.22.19
    $ sequelize db:create --env=production-ssl-disable
    
    Sequelize CLI [Node: 20.10.0, CLI: 6.6.1, ORM: 6.33.0]
    
    Loaded configuration file "server/config/database.json".
    
    ERROR: Dialect undefined does not support db:create / db:drop commands
    
    error Command failed with exit code 1.
    info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
    ```

**Expected behavior**
Execute the db creation without error

**Screenshots**
Not applicable

**Outline (please complete the following information):**
- Install: docker compose in WSL
- Version: docker.getoutline.com/outlinewiki/outline [7e63ee1e9984] but same result with 0.72.0 [2084ece57b76]

**Desktop (please complete the following information):**
 - OS: Ubuntu 20.04.6 LTS (5.10.102.1-microsoft-standard-WSL2) through WSL2 on Windows 11 Pro 23H2
 - Browser: NA
 - Version: NA

I'm trying for hours now and nothing is working like I'm missing something. Could it be related to WSL ?

**Comment by tommoor** [comment]
No idea I'm afraid, the `dialect` key is specified in that file and has been for 7 years without issue. Perhaps it is related to WSL somehow, it is an unusual setup.

**Comment by yseraf** [comment]
My bad... Just a typo issue. Replacing `--env=production-ssl-disable` by `--env=production-ssl-disabled` fixed it.

Thanks @tommoor for the support.
